openstack/blockstorage/extensions/schedulerstats/results.go

```diff
@@ -37,6 +36,25 @@ type StoragePool struct {
 	Capabilities Capabilities `json:"capabilities"`
 }
 
+// Generic function to parse a capacity value which may be a numeric
+// value, "unknown", or "infinite"
+func ParseCapacity(capacity interface{}) float64 {
```

Comment:
I'd prefer to just put this logic in `UnmarshalJSON`, but if you want to use a function, it should to be unexported (lowercase).